### DISCUSSION TITLE
How to view a diff of all the commits by a particular user on GitHub

### DISCUSSION THREAD

**Discussion by reikdas** [community/community#24028] [New to GitHub]
<p>Is there a way to see a diff of all the commits made by a particular user to a particular repository on GitHub?</p>

**Comment by airtower-luna** [comment]
<p>You can use <code>git log</code> with the <a href="https://git-scm.com/docs/git-log#Documentation/git-log.txt---authorltpatterngt"><code>--author</code> option</a> to filter for commits from specific people:</p>
<pre><code class="lang-bash">git log --author=NAME --patch
</code></pre>

**Comment by reikdas** [comment]
<p>I should have specified - I want to view the diff on <a href="http://github.com" rel="noopener nofollow ugc">github.com</a>, not locally using git (and a single diff for all the commits).</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>The closest option on the website that I’m aware of is filtering the commit history of a branch or tag for a specific author. You might be able to search for all commits by a specific person using the API.</p>
<p>I don’t think the single diff part is possible, at least not for most repositories with more than one contributor: Commits by the person you’re interested in might interleave with other people’s commits, and in that case there isn’t really any way to create a single diff of changes only that person made. If you compare the first and latest commits by the person the diff will include changes others made in between. If you try to leave out changes made by others changes from any commit by the person of interest based on those changes won’t fit any more. And if you look at all changes within a repository one person might make conflicting changes on different branches.</p>
